roles/openshift_common/tasks/main.yml

```diff
@@ -1,4 +1,8 @@
 ---
+- fail:
+    msg: Flannel can not be used with openshift sdn
+  when: openshift_use_openshift_sdn | bool and openshift_use_flannel | bool
```

Comment:
To work around the issue @brenton is seeing, the following should do the trick:
when: openshift_use_openshift_sdn | default(false) | bool and openshift_use_flannel | default(false) | bool